File: server/models/project.go
package models

import "time"

type Project struct {
	ID             int64     `json:"id"`
	Name           string    `json:"name"`
	Description    string    `json:"description"`
	Tags           []string  `json:"tags"`
	OwnerID        int64     `json:"ownerId"`
	Owner          *User     `json:"owner,omitempty"`
	ProjectMembers []User    `json:"projectMembers"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

File: dash/src/types/project.ts

import type { User } from './user';

export interface Project {
  id: number;
  name: string;
  description: string;
  tags?: string[];
  ownerId: string | number;
  owner?: User;
  projectMembers: User[];
  createdAt?: string;
  updatedAt?: string;
}

export interface ProjectCreateInput {
  name: string;
  description: string;
  tags: string[];
}

export interface ProjectUpdateInput {
  name: string;
  description: string;
  tags: string[];
}
